### DISCUSSION TITLE
Why do we even have log scale sampling?

### DISCUSSION THREAD

**Discussion by tramchamploo** [optuna/optuna#4048] [Q&A]
I'm guessing it's for parameters with a wide range of scales. However if you perform a simple experiment like this.
![image](https://user-images.githubusercontent.com/708349/194298517-1efdc0a0-b050-4ebb-9b4c-882394ef00d3.png)
It's clearly very skewed towards the lower end. Is this the desired behaviour? Isn't this just a case of recognitive bias?

**Comment by not522** [comment]
Yes, it's the desired behavior of the log-uniform distribution. https://en.wikipedia.org/wiki/Reciprocal_distribution
If you use the uniform distribution, smaller values are rarely sampled. For example, the probability of sampling a value less than 1 is 0.01% in your case. The log-uniform distribution improves the bias.
